mod/state-transition/pkg/core/state_processor_staking.go

```diff
@@ -40,26 +40,46 @@ func (sp *StateProcessor[
 	st BeaconStateT,
 	blk BeaconBlockT,
 ) error {
-	// Verify that outstanding deposits are processed up to the maximum number
-	// of deposits.
-	deposits := blk.GetBody().GetDeposits()
-	index, err := st.GetEth1DepositIndex()
+	// Verify that outstanding deposits matches those listed by contract
+	depositIndex, err := st.GetEth1DepositIndex()
 	if err != nil {
 		return err
 	}
-	eth1Data, err := st.GetEth1Data()
+
+	stateDeposits, err := sp.ds.GetDepositsByIndex(
+		depositIndex+1, // pick expected next deposit index
```

Comment:
coherent with block_builder implementation, we expect to process in this block the next deposit following latest one (whose index is returned by `GetEth1DepositIndex`)